Add a sample onnx_to_torch lit test

ljfitz opened the pull request:
I took a test that Ramon shared with me and tried to make some minimal changes to make it a `lit` test within `onnx-mlir.

With this file, `ninja onnx-mlir-test` will include this test among any others within `third-party/onnx-mlir/test/mlir`. 
Once this has run once (to set up the configuration) then `$FLEXML_HOME/third-party/llvm-project/./bin/llvm-lit -sv $FLEXML_HOME/third-party/onnx-mlir/test/mlir/conversion/onnx_to_torch` should run just the `onnx_to_torch` `lit` tests.

jcamacho1234 commented:
I've recently pushed in changes to the ONNXToTorch passes where symmetric padding is properly handled in the ATen output. With these new changes, this test will fail. Would you prefer to update the test before merging as opposed to getting the PR approved first and then changing the test?

ljfitz commented:
> I've recently pushed in changes to the ONNXToTorch passes where symmetric padding is properly handled in the ATen output. With these new changes, this test will fail. Would you prefer to update the test before merging as opposed to getting the PR approved first and then changing the test?

Thanks @jcamacho1234 for the pointer.I've taken your updates in and adjusted the test to use the corrected values. I saw you also made a change that made it easy to get onnx-mlir from the ONNX model so I've also adjusted the test to use onnx-mlir as an input which seems more appropriate for testing onnx-mlir to torch-mlir conversion.

JRUMacDev commented:
Liam - this looks very good. Thank you for putting together the working test.

ljfitz commented:
Feedback seems all positive - I'll merge  this PR.

ayonam-helprack reviewed (approved):
This looks otherwise fine.

Don't you think we should use CHECK-NEXT for the Dilation, Pad and the Conv2D operation?  That way we can be sure of the generation of these operations in a certain order also.

ljfitz commented:
> This looks otherwise fine.
> 
> Don't you think we should use CHECK-NEXT for the Dilation, Pad and the Conv2D operation? That way we can be sure of the generation of these operations in a certain order also.

`CHECK-NEXT` could be used, and would mean that no ops should be allowed in between. That seems unnecessarily strict here, as the match for the `conv2d` is checking that the right values are being used in the right place.

I was tempted to use `CHECK-DAG` to indicate that the order of list creation was really unimportant.

But if you want the strict "there can be nothing between" checking then `CHECK-NEXT` is the right tool.

Note if your pass could be made to work in `onnx-mlir-opt` then you get some additional testing support. Last time I tried though the pass setup didn't work with the `onnx-mlir-opt` pass manager :-(

ayonam-helprack commented:
> > This looks otherwise fine.
> > Don't you think we should use CHECK-NEXT for the Dilation, Pad and the Conv2D operation? That way we can be sure of the generation of these operations in a certain order also.
> 
> `CHECK-NEXT` could be used, and would mean that no ops should be allowed in between. That seems unnecessarily strict here, as the match for the `conv2d` is checking that the right values are being used in the right place.
> 
> I was tempted to use `CHECK-DAG` to indicate that the order of list creation was really unimportant.
> 
> But if you want the strict "there can be nothing between" checking then `CHECK-NEXT` is the right tool.
> 
> Note if your pass could be made to work in `onnx-mlir-opt` then you get some additional testing support. Last time I tried though the pass setup didn't work with the `onnx-mlir-opt` pass manager :-(

Yes, I agree.  

I was actually hoping to check the correct execution of the multiple steps in the pass using the CHECK-NEXT.  But I agree, CHECK-NEXT would be too strict since another pass after our transformation pass could change the order of the statements and then this test would fail.  And besides, the matcher in the Conv2D is indeed checking the operands to be in the right places.  So the semantic equivalence is well checked.

We will try to get the pass to work with onnx-mlir-opt pass manager.  Let us finish the current set of operations first.  Flatten is still an open item for us.  The rest are done though.
